vendor/github.com/docker/distribution/registry/storage/linkedblobstore.go

```diff
@@ -289,11 +281,19 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 	return nil
 }
 
-func (lbs *linkedBlobStore) mount(ctx context.Context, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
+func (lbs *linkedBlobStore) mount(ctx context.Context, repoOverride distribution.RepositoryOverride, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
 	repo, err := lbs.registry.Repository(ctx, sourceRepo)
 	if err != nil {
 		return distribution.Descriptor{}, err
 	}
+
+	if repoOverride != nil {
+		repo, err = repoOverride(ctx, repo)
```

Comment:
this seems really heavyweight... why wouldn't we just add an optional `Stat *distribution.Descriptor` field to the `Mount` struct? If present, the repo/blobs/stat calls are not needed at all